Migrate "Make class abstract" to driver interface

hernanmd opened the pull request:
Re-worked the implementation so it does not send `realClass`.
Use ReClassesHaveNoReferencesCondition to check.
Clean RBMakeClassAbstractTransformation menu item string.
Use #RBClassWithoutSelfClassReferenceTest and #RBWithSelfClassReferenceTest to test the transformation.
Add convenience method `model` to interaction driver.
Add comments.
Add tests.

Commit:
First set of commits for Make Abstract Transformation Driver
Add test.

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into make_class_abstract_driver

Commit:
Re-worked the implementation so it does not send `realClass`.
Use ReClassesHaveNoReferencesCondition to check.
Clean RBMakeClassAbstractTransformation menu item string.
Use #RBClassWithoutSelfClassReferenceTest and #RBWithSelfClassReferenceTest to test the transformation.
Add convenience method `model` to interaction driver.
Add comments.
Add tests.

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into make_class_abstract_driver

Commit:
WiP

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into make_class_abstract_driver

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into add_subclass_refactoring_driver

Ducasse commented:
Strange we have extra tests failing

```
osx-64 / Tests-osx-64 / testClassVariable – MacOSX64.Refactoring.Transformations.Tests.RBRemoveDirectAccessToVariableTransformationTest
```

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into make_class_abstract_driver

Commit:
Merge e8470c7a6a58d4289a8a7640674d8ab86dccf6e9

Commit:
Merge db7e27ae0cc9f710d3ea04d1ef96a506a0a55bcd

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into make_class_abstract_driver

Commit:
Migrate the RBMakeClassAbstractParametrizedTest tests.
Unreference classWithoutReferenceToAbstract

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into add_subclass_refactoring_driver

hernanmd committed:
Merge branch 'pharo-project:Pharo12' into make_class_abstract_driver

Commit:
Ported #RBMakeClassAbstractTransformation to the new driver interface.
Add ReMakeClassAbstractDriverTest and add classes to Refactoring-DataForTesting
Add #RBMakeClassAbstractParametrizedTest.
Fix privateTransform in RBAddMethodTransformation
Updated SystemCommands for Make Abstract

Commit:
Make abstract class driver migration
